nmostesting/suites/IS1102Test.py

```diff
@@ -13,16 +13,1402 @@
+                    if color_sampling != new_color_sampling:
+                        return test.FAIL("different color_sampling")
+                if item == "urn:x-nmos:cap:format:component_depth":
+                    if (
+                        flow["components"][0]["bit_depth"]
+                        != new_flow["components"][0]["bit_depth"]
+                    ):
+                        return test.FAIL("different component_depth")
+
+            _, response = TestHelper.do_request(
+                "DELETE",
+                self.compat_url + "senders/" + sender_id + "constraints/active/",
+            )
+            if response.status_code != 200:
+                return test.FAIL(response.json())
+        return test.PASS()
+
+    def test_02_02_06_02(self, test):
+        """Verify that setting NOP constraints for supported
+        constraints does not change the flow of  a sender (audio)
+        and that the state goes from \"unconstrained\" to \"constrained\"
+        """
+        if len(self.flow_format_audio) == 0:
+            return test.UNCLEAR("There is no audio format")
+        for sender_id in self.flow_format_audio:
+            _, response = TestHelper.do_request(
+                "GET", self.compat_url + "senders/" + sender_id + "status/"
+            )
+            if response.status_code != 200:
+                test.FAIL(response.json())
+            state = response.json()["state"]
+            if state in ["awating_essence", "no_essence"]:
+                for i in range(0, 5):
+                    _, response = TestHelper.do_request(
+                        "GET", self.compat_url + "senders/" + sender_id + "status/"
+                    )
+                    state = response.json()["status"]["state"]
+                    if state in ["awating_essence", "no_essence"]:
+                        time.sleep(3000)
+                    else:
+                        break
+            if state != "unconstrained":
+                return test.FAIL("inputs are unstable.")
+            _, response = TestHelper.do_request(
+                "GET", self.node_url + "senders/" + sender_id
+            )
+            if response.status_code != 200:
+                return test.FAIL(response.json)
+            sender = response.json()
+
+            _, response = TestHelper.do_request(
+                "GET", self.node_url + "flows/" + sender["flow_id"]
+            )
+            if response.status_code != 200:
+                return test.FAIL(response.json())
+            flow = response.json()
+            constraint_set = {}
+
+            _, response = TestHelper.do_request(
+                "GET", self.node_url + "sources/" + flow["source_id"]
+            )
+            if response.status_code != 200:
+                return test.FAIL(response.json())
+            source = response.json()
+
+            for item in REF_SUPPORTED_CONSTRAINTS_AUDIO:
+
```

Comment:
Should put the body of ALL the for loop using REF_SUPPORTED_* between try: except: pass

The reason is that a Sender indicates the supported constraints based on its "format" which is video or audio. But because in NMOS some Flow properties are specific to the media_type being raw or coded, some constraints like "sample_depth" for example would would not apply to a coded audio flow because the associated" bit_depth" attribute is not available for a coded audio flow. The same applies to video raw versus coded.

So everywhere the construct "for item in REF_SUPPORTED_CONSTRAINTS_*" is used have the body of the loop between"try:body except:pass" such that if a attribute is not available in the Flow, it will not crash the test that will silently ignore it as it should be.